api/app/controllers/spree/api/v2/storefront/products_controller.rb

```diff
@@ -46,6 +46,14 @@ def scope_includes
           def allowed_sort_attributes
             super << :available_on
           end
+
+          def collection_meta(collection)
+            super(collection).merge(filters: filters_meta)
+          end
+
+          def filters_meta
+            Spree::Api::Products::FiltersPresenter.new(current_store, current_currency, params).to_h
```

Comment:
I think this should really be a serializer, not a presenter. We can of course reuse presenters inside that serializer - like it's already done. What do you think?

Reply:
@mad-eel let's stick with the presenter for now as this will lead to some inconsistencies - we only have JSON::API serializer currently

Reply:
I was thinking the same - the JSON API serializers output more complex structure which is not needed here (and I'm not sure if we can actually put that in meta)